homework1/Lab2.js

```diff
@@ -0,0 +1,394 @@
+                    }
+                }
+                this.store.foods = updated_list;
+            }
+            this.render();
+    }
+
+    render () {
+        // using innerHTML to render inventory listing
+        let foodbody=this.root.querySelector('tbody');
+        foodbody.innerHTML='';
+        if(this.store.foods===null){
+            this.store.foods=[];
+        }
+        for(var i=0;i<this.store.foods.length;i++){
+            var food_name=this.store.foods[i].name;
+            var food_img=this.store.foods[i].image;
+            var food_description=this.store.foods[i].description;
+
+            foodbody.innerHTML +=
+           
+                    `<tr >
+                     <td>${food_name}</td>
+                        <td>
+                            <img src="${food_image}">
+                        </td>
+                        <td>${food_description}
+                        </td>
+                        <td>
+                            <button class="removebutton" data-index=${i}>Remove item from Inventory</button>
+                         </td>
+                      </tr>`;
+        }
+        let removeButton=document.querySelectorAll('.removebutton');
+        for(var i=0;i<removeButton.length;i++){
+            let buttn=removeButton[i];
+            buttn.addEventListener('click',()=>{
+                let item=this.store.foods[parseInt(buttn.dataset.index)];
+                this.removeItem(item);
+            }
+                );
+            }
+        }
+    }
+
+
+class Menu {
+    constructor(root, store) {
+        this.root = root;
+        this.store = store;
+        this.init();
+    }
+
+    init () {
+        this.render();
+    }
+
+    render () {
+        // TODO: render a list of food menu from store using innerHTML
+    // using innerHTML to render inventory listing
+            let tbody = this.root.querySelector('tbody');
+            // using innerHTML to render a list of table row item under tbody
+            tbody.innerHTML = ``;
+            if (this.store.foods === null) {
+                this.store.foods = [];
+            } 
+            for (var i = 0; i < this.store.foods.length; i++) {
+                // for each item in local storage's FOODS, create a row with a cell for the item name and image, and one for description.
+                var food_name = this.store.foods[i].name;
+                var food_image = this.store.foods[i].image;
+                var food_description = this.store.foods[i].description;
+
+                tbody.innerHTML +=
+                    `<tr >
+                     <td>${food_name}</td>
+                        <td>
+                            <img src=${food_image}>
+                        </td>
+                        <td>${food_description}
+                        </td>
+                        <td>
+                            <button class="checkout-button" data-index=${i}>Add item to cart</button>
```

Comment:
You will need more data attribute like name and price unless you modify the implementation of the CheckoutButton (which I didn't see any modification).

In other word, the code here wouldn't work with the checkout button.